pallets/messages/runtime-api/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::unnecessary_mut_passed)]
+
+use codec::Codec;
+use common_primitives::messages::*;
+use frame_support::dispatch::DispatchError;
+
+// Here we declare the runtime API. It is implemented it the `impl` block in
+// runtime file (the `runtime/src/lib.rs`)
+sp_api::decl_runtime_apis! {
+	pub trait MessagesApi<AccountId, BlockNumber> where
+		AccountId: Codec,
+		BlockNumber: Codec,
+	{
+		fn get_messages_by_schema(schema_id: SchemaId,
+			pagination: BlockPaginationRequest<BlockNumber>) ->
+		Result<
+			BlockPaginationResponse<BlockNumber, MessageResponse<AccountId, BlockNumber>>,
+			DispatchError,
+		>;
+	}
```

Comment:
How are we ensuring it is signed and paid for? I been thinking how we can execute an extrinsic via rpc (to do small offchain stuff)
Like schemas will submitted to an rpc that will quickly validated and submit to register_schema extrinsic instead of public function

Reply:
That is a good question. So read-only RPC's like this one don't need to be signed or paid for IMO since it is not writing any data on-chain. We only need some kind of `quota limits` implementation using offchain db to ensure this is not going to get abused.

All extrinsics are queued using the following. I think looking into how this is implemented is going to be useful
```
	#[rpc(name = "author_submitExtrinsic")]
	fn submit_extrinsic(&self, extrinsic: Bytes) -> FutureResult<Hash>;
```